pkg/virtual/workspaces/registry/rest.go

```diff
@@ -269,32 +269,31 @@ func (s *REST) deprecatedAuthorizeOrgForUser(ctx context.Context, orgClusterName
 }
 
 func (s *REST) authorizeForUser(ctx context.Context, orgClusterName logicalcluster.Name, currentUser kuser.Info, verb string, resourceName string) error {
-	// Root org access is implicit for every user. For non-root orgs, we need to check for
-	// verb permissions against the clusterworkspaces/workspace sub-resource.
-	if orgClusterName == tenancyv1alpha1.RootCluster || sets.NewString(currentUser.GetGroups()...).Has(kuser.SystemPrivilegedGroup) {
+	if sets.NewString(currentUser.GetGroups()...).Has(kuser.SystemPrivilegedGroup) {
 		return nil
 	}
 
-	// Maintain compatibility with the old permission model until we completel remove
-	// the ability to have personal workspaces in top-level organizations
-	if parent, _ := orgClusterName.Split(); parent == tenancyv1alpha1.RootCluster {
-		switch verb {
-		case "get", "watch", "list":
-			return s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, currentUser, "access")
-		case "create":
-			return s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, currentUser, "member")
-		case "delete":
-			if err := s.deprecatedAuthorizeOrgForUser(ctx, orgClusterName, currentUser, "access"); err != nil {
-				return err
+	if orgClusterName != tenancyv1alpha1.RootCluster {
```

Comment:
what if `orgClusterName` == `tenancyv1alpha1.RootCluster` ?

Reply:
RootCluster has no parent. Hence we cannot check anything in there. Before this PR we skipped the whole authz step though 😱. Now we only skip the access/member check in the parent.

Reply:
do we have separate checks for users targeting only the `RootCluster`?

Reply:
added the tests.

The logic for root is that everybody has implicit `access` permissions, i.e. can "enter" and has the group membership of `system:kcp:clusterworkspace:access`.

Reply:
nit: is this `if` statement necessary? The logic inside only executes if orgClusterName's parent is root. That logic works without checking if orgClusterName != root.... right?

Reply:
it's not necessary. True.